js/artoolkit.multi_worker.js

```diff
@@ -0,0 +1,61 @@
+importScripts('../build/artoolkit.min.js');
+
+self.onmessage = function(e) {
+    var msg = e.data;
+    switch (msg.type) {
+        case "load": {
+            load(msg);
+            return;
+        }
+        case "process": {
+            next = msg.imagedata;
+            process();
+            return;
+        }
+    }
+};
+
+var next = null;
+
+var ar = null;
+var markerResult = null;
+
+function load(msg) {
+
+    var param = new ARCameraParam(msg.camera_para);
+
+    param.onload = function () {
+        ar = new ARController(msg.pw, msg.ph, param);
+        var cameraMatrix = ar.getCameraMatrix();
+
+        ar.addEventListener('getNFTMarker', function (ev) {
+            console.log(ev);
+            markerResult = {type: "found", index: JSON.stringify(ev.data.index),  matrixGL_RH: JSON.stringify(ev.data.matrixGL_RH), proj: JSON.stringify(cameraMatrix)};
+        });
+
+        ar.loadNFTMarkers(msg.marker, function (markerId) {
```

Comment:
what's the callback `markerId` here?

Reply:
given the markerId it tracks the marker, we have already used this pattern.

Reply:
msg.marker seems to be a list. the markerId refers to the whole list rather than an individual marker?

Reply:
yes it is refering to the array/list of markers

Reply:
printing markerId, it will give you the whole list

Reply:
Thinking now, i could ahnge this to track every single marker inside this routine, and esnding a postmessage for every single marker? What is better?

Reply:
i will do a snippet, as example.

Reply:
We have trhee NTF markers, so i can track each one? Do you think it is more correct this code?

```javascript
ar.loadNFTMarkers(msg.marker, function (markerId) {
            ar.trackNFTMarkerId(markerId[0], 2);
            ar.trackNFTMarkerId(markerId[1], 2);
            ar.trackNFTMarkerId(markerId[2], 2);
            console.log("loadNFTMarker -> ", markerId);
            postMessage({type: "endLoading", end: true})
        });
```

Reply:
I see. I think what you have originally is fine. You probably want to track all markers anyway. Perhaps, naming them `msg.makers` and `markerIds` can avoid some confusion. But it's just a minor issue. 

I have no problem with this PR. I haven't tried running the example, but the code looks fine with me. 

Thanks for the effort!

Reply:
@hiukim thanks for the review! i will think about this last considerations, and yes you are right i will rename with a more convenient name.

Reply:
@hiukim at the end i changed the names... :slightly_smiling_face: